### PR TITLE
Fixes #890. Remove intermittent failing search tests.

### LINKED ISSUE #890: Intermittent Search test failures
This fails pretty frequently, but not 100% of the time. And if you re-trigger a build on Travis it will usually pass after the failure.

So... it would be good to figure out why this happens (and why it's a recent thing).

```
FAIL: main - Search (non-auth) - Search works (3496ms)

AssertionError: The search results show up on the page.: expected 'Issue 280: w3.org - desktop site instead of mobile site' to include 'vladvlad'

  at Function.assert.include  <node_modules/intern/node_modules/chai/chai.js:2977:45>

  at CompatCommand.<anonymous>  <tests/functional/search-non-auth.js:92:18>

  at runCallback  <node_modules/intern/node_modules/leadfoot/Command.js:509:31>

  at CompatCommand.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:526:11>

  at <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:235:37>

  at runCallbacks  <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:11:18>

  at <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:188:17>

  at Object.run [as _onImmediate]  <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:131:29>

  at processImmediate [as _immediateCallback]  <timers.js:354:15>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:525:10>

  at Test.registerSuite.Search works [as test]  <tests/functional/search-non-auth.js:91:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.Command.end  <node_modules/intern/node_modules/leadfoot/Command.js:475:10>

  at Test.registerSuite.Search works [as test]  <tests/functional/search-non-auth.js:94:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

127.0.0.1 - - [30/Dec/2015 02:19:03] "GET / HTTP/1.1" 200 -

Expected fixture file: /home/travis/build/webcompat/webcompat.com/tests/fixtures/api/issues/labels.6665c4820456e6a813dcbbdddd690ffc.json

127.0.0.1 - - [30/Dec/2015 02:19:03] "GET /api/issues/labels?per_page=100 HTTP/1.1" 200 -

127.0.0.1 - - [30/Dec/2015 02:19:03] "GET /api/issues/category/new HTTP/1.1" 200 -

127.0.0.1 - - [30/Dec/2015 02:19:04] "GET /issues?q=vladvlad HTTP/1.1" 200 -

Expected fixture file: /home/travis/build/webcompat/webcompat.com/tests/fixtures/api/issues/labels.6665c4820456e6a813dcbbdddd690ffc.json

127.0.0.1 - - [30/Dec/2015 02:19:04] "GET /api/issues/labels?per_page=100 HTTP/1.1" 200 -

FAIL: main - Search (non-auth) - Search from the homepage (14080ms)

NoSuchElement: [POST http://localhost:4444/wd/hub/session/a82eeaae-3abd-4880-a73f-bab8a12c0b58/element / {"using":"css selector","value":".wc-IssueItem:nth-of-type(1) a"}] Unable to locate element: {"method":"css selector","selector":".wc-IssueItem:nth-of-type(1) a"}

For documentation on this error, please visit: http://seleniumhq.org/exceptions/no_such_element.html

Build info: version: '2.48.0', revision: 'b7b081a', time: '2015-10-07 15:48:56'

System info: host: 'testing-worker-linux-docker-8399f37a-3194-linux-5', ip: '172.17.1.255', os.name: 'Linux', os.arch: 'amd64', os.version: '3.13.0-40-generic', java.version: '1.7.0_76'

Driver info: driver.version: unknown

  at Server._post  <node_modules/intern/node_modules/leadfoot/Server.js:68:9>

  at ProxiedSession._post  <node_modules/intern/node_modules/leadfoot/Session.js:57:30>

  at ProxiedSession.Session.find  <node_modules/intern/node_modules/leadfoot/Session.js:1053:15>

  at CompatCommand.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:41:36>

  at <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:235:37>

  at runCallbacks  <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:11:18>

  at <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:188:17>

  at Object.run [as _onImmediate]  <node_modules/intern/node_modules/leadfoot/node_modules/dojo/Promise.js:131:29>

  at processImmediate [as _immediateCallback]  <timers.js:354:15>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.find  <node_modules/intern/node_modules/leadfoot/Command.js:22:10>

  at <CompatCommand.prototype.(anonymous function) [as findByCssSelector] (/home/travis/build/webcompat/webcompat.com/node_modules/intern/node_modules/leadfoot/lib/strategies.js:24:16)>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:108:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at <CompatCommand.target.(anonymous function) [as getVisibleText] (/home/travis/build/webcompat/webcompat.com/node_modules/intern/node_modules/leadfoot/Command.js:656:11)>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:108:62>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:525:10>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:109:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.Command.end  <node_modules/intern/node_modules/leadfoot/Command.js:475:10>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:112:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at <CompatCommand.target.(anonymous function) [as getCurrentUrl] (/home/travis/build/webcompat/webcompat.com/node_modules/intern/node_modules/leadfoot/Command.js:592:11)>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:113:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.Command.then  <node_modules/intern/node_modules/leadfoot/Command.js:525:10>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:114:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

  at new CompatCommand  <node_modules/intern/runner.js:213:14>

  at CompatCommand.Command.end  <node_modules/intern/node_modules/leadfoot/Command.js:475:10>

  at Test.registerSuite.Search from the homepage [as test]  <tests/functional/search-non-auth.js:117:10>

  at Test.run  <node_modules/intern/lib/Test.js:169:19>

  at <node_modules/intern/lib/Suite.js:237:13>

  at signalListener  <node_modules/intern/node_modules/dojo/Deferred.js:37:21>

  at Promise.then.promise.then  <node_modules/intern/node_modules/dojo/Deferred.js:258:5>

  at runTest  <node_modules/intern/lib/Suite.js:236:46>

  at <node_modules/intern/lib/Suite.js:249:7>

  at process._tickCallback  <node.js:442:13>

2/6 tests failed
```

### PR DESCRIPTION
**Pull request by miketaylr**
It dawned on me that the reason these sometimes fail is because of Search API rate limits (non-authed tests are sending requests from the Travis IP, not the authed webcompat proxy).

If that theory is true, then moving the "search works" to the auth tests should never fail (which is good, because we don't lose test coverage -- sort of).

r? @karlcow

### CODE REVIEW

**Comment by miketaylr** [comment]
Erm.

```
FAIL: main - Search (auth) - Search works (3577ms)

AssertionError: The search results show up on the page.: expected 'Issue 280: w3.org - desktop site instead of mobile site' to include 'vladvlad'
```

**Comment by karlcow** [comment]
ok I have 3 failed tests.

capture d ecran 2016-04-29 a 12 09 13

Let see:

```
× firefox 45.0.2 on MAC - Issues (auth) - Closing and reopening an issue
× firefox 45.0.2 on MAC - Reporting (non-auth) - Problem type validation
× firefox 45.0.2 on MAC - Reporting (non-auth) - URL validation
```

hmm not related to your modifications at least.
Let me try again.

grmph

capture d ecran 2016-04-29 a 12 15 58


```
× firefox 45.0.2 on MAC - Issues (auth) - Closing and reopening an issue
AssertionError: Closed state text is displayed: expected 'Ready for Outreach' to equal 'Closed'
× firefox 45.0.2 on MAC - Search (auth) - Search works
AssertionError: The search results show up on the page.: expected 'Issue 369: www.testing-things.com - mobile site is not usable' to include 'vladvlad'
```

**Comment by karlcow** [comment]
so unstable these tests… Sniff. 💣

**Comment by miketaylr** [comment]
> so unstable these tests… Sniff. 💣

Yeah. It really sucks. :(

**Comment by miketaylr** [comment]
> ok I have 3 failed tests.

And yet Travis likes all the tests: https://travis-ci.org/webcompat/webcompat.com/builds/126474683

Feels like playing Russian roulette. 🔫

**Comment by karlcow** [comment]
ok let's merge. Something to discuss in London or somewhere. In the same bucket of fixtures and Mocking.

**Comment by karlcow** [comment]
Thanks @miketaylr 💡
